source/middleware/LoggingMiddleware.ts

```diff
@@ -0,0 +1,22 @@
+import type { Request, Response, NextFunction } from 'express';
+import { UserLog, type Log } from '../models/index.js';
+import { v4 as uuidv4 } from 'uuid';
+import { logger } from '../lib/Logger.js';
+
+export async function logUserAction (req: Request, res: Response, next: NextFunction): Promise<void> {
+	if (res.locals.user !== undefined && res.locals.user !== null) {
+		logger.log(`User ${res.locals.user.username} made ${req.method} to ${req.path}!`);
+		await UserLog.create({
+			id: uuidv4(),
+			log: { action: req.method, path: req.path } satisfies Log,
+			userId: res.locals.user?.id
+		});
+	} else {
+		logger.log(`Anonymous made ${req.method} to ${req.path}!`);
+		await UserLog.create({
+			id: uuidv4(),
+			log: { action: req.method, path: req.path } satisfies Log
+		});
+	}
```

Comment:
Please encase this in a try catch block to ensure that a failure to add logs to database cannot impede progress.
Also, I see no reason to `await` for the user log row to be generated, thus no need for the function to be async

Reply:
Done, [6510386](https://github.com/Scarpy019/TTII/pull/21/commits/65103860ffc821131d9056011d36aa90a133bd6a).